tck/features/PatternPredicateAcceptance.feature

```diff
@@ -0,0 +1,248 @@
+      """
+      CREATE ({node: 1})-[:X {rel: 1}]->({node: 2})
+      CREATE ({node: 3})-[:X {rel: 2}]->({node: 4})
+      """
+    When executing query:
+      """
+      MATCH (n) WHERE NOT (n)-[{rel: 1}]->()
+      RETURN n.node AS id
+      """
+    Then the result should be:
+      | id |
+      | 2  |
+      | 3  |
+      | 4  |
+    And no side effects
+
+  Scenario: Filter using a variable length relationship pattern predicate with properties
+    And having executed:
+      """
+      UNWIND [{node: 12, rel: 42}, {node: 324234, rel: 666}] AS props
+      CREATE (:START {p: props.node})-[:X {prop: props.rel}]->()-[:X {prop: props.rel}]->()
+      """
+    When executing query:
+      """
+      MATCH (n:START) WHERE (n)-[*2 {prop: 42}]->()
+      RETURN n.p AS p
+      """
+    Then the result should be:
+      | p  |
+      | 12 |
+    And no side effects
+
+  Scenario: Filter using a pattern predicate that is a logical OR between an expression and a subquery
+    And having executed:
+      """
+      UNWIND [{node: 33, rel: 42}, {node: 12, rel: 666}, {node: 55555, rel: 7777}] AS props
+      CREATE (:START {p: props.node})-[:X {prop: props.rel}]->()-[:X {prop: props.rel}]->()
+      """
+    When executing query:
+      """
+      MATCH (n:START) WHERE n.p = 12 OR (n)-[*2 {prop: 42}]->()
+      RETURN n.p AS p
+      """
+    Then the result should be:
+      | p  |
+      | 33 |
+      | 12 |
+    And no side effects
+
+  Scenario: Filter using a pattern predicate that is a logical OR between two expressions and a subquery
+    And having executed:
+      """
+      UNWIND [{node: 33, rel: 42}, {node: 12, rel: 666}, {node: 25, rel: 444}, {node: 55555, rel: 7777}] AS props
+      CREATE (:START {p: props.node})-[:X {prop: props.rel}]->()-[:X {prop: props.rel}]->()
+      """
+    When executing query:
+      """
+      MATCH (n:START) WHERE n.p = 12 OR (n)-[*2 {prop: 42}]->() OR n.p = 25
+      RETURN n.p AS p
+      """
+    Then the result should be:
+      | p  |
+      | 33 |
+      | 12 |
+      | 25 |
+    And no side effects
+
+  Scenario: Filter using a pattern predicate that is a logical OR between one expression and a negated subquery
+    And having executed:
+      """
+      UNWIND [{node: 25, rel: 444}, {node: 12, rel: 42}, {node: 25, rel: 42}] AS props
+      CREATE (:START {p: props.node})-[:X {prop: props.rel}]->()-[:X {prop: props.rel}]->()
+      """
+    When executing query:
+      """
+      MATCH (n:START) WHERE n.p = 12 OR NOT (n)-[*2 {prop: 42}]->()
+      RETURN n.p AS p
+      """
+    Then the result should be:
+      | p  |
+      | 25 |
+      | 12 |
+    And no side effects
+
+  Scenario: Filter using a pattern predicate that is a logical OR between one subquery and a negated subquery
+    And having executed:
+      """
+      CREATE (s1:START {id: 1})
+      CREATE (s1)-[:X]->({prop: 42})
+      CREATE (s2:START {id: 2})
+      CREATE (s2)-[:X]->({prop: 411})
+      CREATE (s3:START {id: 3})
```

Comment:
I like to separate creation of nodes and relationships and use `CREATE` as a separator for the two 'phases', like so:
```
CREATE (s1:START {id: 1}), (s2:START {id: 2}), (s3:START {id: 3})
CREATE (s1)-[:X]->({prop: 42}),
       (s2)-[:X]->({prop: 411})
```

This is fairly minor and not crucial, but I wanted to share.